dojo/models.py

```diff
@@ -526,6 +527,20 @@ def get_numerical_severity(severity):
         else:
             return 'S4'
 
+    def legacy_save(self, *args, **kwargs):
+        super(Finding, self).save(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        super(Finding, self).save(*args, **kwargs)
```

Comment:
Instead of calling self.legacy_save() on 542, you should call this at the bottom of save(), outside the check for ENABLE_DEDUPLICATION. I think we can do without legacy_save() entirely since it's not called anywhere else in the codebase, and the below lines would only affect the user if they explicitly enable deduplication. I can't think of a reason we'd want to ignore the ENABLE_DEDUPLICATION setting if the user potentially set it to True.